src/mimemail.erl

```diff
@@ -1147,15 +1156,30 @@ dkim_hash_data(CanonicHeaders, DkimHeader) ->
 	JoinedHeaders = << <<Hdr/binary, "\r\n">> || Hdr <- CanonicHeaders>>,
 	crypto:hash(sha256, <<JoinedHeaders/binary, DkimHeader/binary>>).
 
-dkim_sign(DataHash, {pem_plain, PrivBin}) ->
+dkim_get_algorithm_digest(Algorithm) ->
+	case Algorithm of
+		'rsa-sha256' -> sha256;
+		'ed25519-sha256' ->
+			% %% TODO: Remove `public_key` vsn check once we require Erlang/OTP 24.1+
+			{ok, PublicKeyAppVersionString} = application:get_key(public_key, vsn),
+			PublicKeyAppVersionList = lists:map(fun erlang:list_to_integer/1, string:tokens(PublicKeyAppVersionString, ".")),
+			
+			if
+				PublicKeyAppVersionList < [1, 11, 2] -> throw("DKIM with Ed25519 requires Erlang/OTP 24.1+");
+				true -> none
+			end
+	end.
+
+dkim_sign(DataHash, Algorithm, {pem_plain, PrivBin}) ->
 	[PrivEntry] = public_key:pem_decode(PrivBin),
+	Digest = dkim_get_algorithm_digest(Algorithm),
 	Key = public_key:pem_entry_decode(PrivEntry),
-	public_key:sign({digest, DataHash}, sha256, Key);
-dkim_sign(DataHash, {pem_encrypted, EncPrivBin, Passwd}) ->
+	public_key:sign({digest, DataHash}, Digest, Key);
+dkim_sign(DataHash, Algorithm, {pem_encrypted, EncPrivBin, Passwd}) ->
 	[EncPrivEntry] = public_key:pem_decode(EncPrivBin),
+	Digest = dkim_get_algorithm_digest(Algorithm),
 	Key = public_key:pem_entry_decode(EncPrivEntry, Passwd),
```

Comment:
Might be cool to have a test checking that encrypted private keys work for `Ed25519` keys as well

Reply:
(but I have an impression that I was a comment somewhere saying that encrypted `Ed25519` are not supported yet